lang/py/test/test_schema.py

```diff
@@ -397,13 +447,22 @@ def __init__(self, test_schema):
     super(SchemaParseTestCase, self).__init__(
         'parse_valid' if test_schema.valid else 'parse_invalid')
     self.test_schema = test_schema
+    # Never hide repeated warnings when running this test case.
+    warnings.simplefilter("always")
```

Comment:
Quick note -- without this line, some of the tests expecting warnings will fail.  I'm assuming the "default" filter mode has the desired behaviour of only registering the warning the first time it is encountered!

Reply:
The default filter will print the warning once for each location it’s encountered in. There is also a “once” filter.

Reply:
Quick question -- is there something to do or test to have the "once" filter, or is the default OK for `IgnoredLogicalType` exceptions?  I'm taking a closer look at the filtering...

Reply:
I am not sure I understand the question, but I think as long as you're testing that the warnings.warn is being called, you can trust the implementation of warnings underneath to do the right thing.

Reply:
Ah... from the JIRA: `I think the warning should be enabled in "once" mode by default.`

I think I half-understood, and I was trying to use the technique [here](https://docs.python.org/3/library/warnings.html#overriding-the-default-filter) to change the filter action to `once` if the application hadn't set a different filter action in their application.

I'll leave the implementation underneath and the default behaviour (once per location).